### ISSUE TITLE
[fontconfig] build failure on windows 10

### ISSUE THREAD

**Issue by Baklap4** [microsoft/vcpkg#19419]
**Host Environment**
- OS: Windows 10
- Compiler: 2021-07-26-9425cf5f512f242c0bcbabac31f08832825aee81 
- vcpkg repo revision: 5742e878fc71b4fcb9c019b50f1f887f22db58df

**To Reproduce**
`./vcpkg install fontconfig`

**Failure logs** 
Powershell logs:
```
Starting package 15/15: fontconfig:x64-windows
Building package fontconfig[core]:x64-windows...
-- Using C:/Users/Arjan/source/repos/performous/vcpkg/downloads/fontconfig-fontconfig-844d8709a1f3ecab45015b24b72dd775c13b2421.tar.gz
-- Cleaning sources at C:/Users/Arjan/source/repos/performous/vcpkg/buildtrees/fontconfig/src/75c13b2421-7d202e6ff4.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Users/Arjan/source/repos/performous/vcpkg/downloads/fontconfig-fontconfig-844d8709a1f3ecab45015b24b72dd775c13b2421.tar.gz
-- Applying patch remove_tests.patch
-- Applying patch build.patch
-- Applying patch build2.patch
-- Applying patch disable-install-data.patch
-- Applying patch macos_arm_fccache.patch
-- Applying patch fix_def_dll_name.patch
-- Using source at C:/Users/Arjan/source/repos/performous/vcpkg/buildtrees/fontconfig/src/75c13b2421-7d202e6ff4.clean
-- Found external ninja('1.10.2').
-- Getting CMake variables for x64-windows
-- Using msys root at C:/Users/Arjan/source/repos/performous/vcpkg/downloads/tools/msys2/2f5e2150b81fcd06
-- Using msys root at C:/Users/Arjan/source/repos/performous/vcpkg/downloads/tools/msys2/aa5af7b2aa7e90e8
-- Generating configure for x64-windows
-- Finished generating configure for x64-windows
-- Configuring x64-windows-dbg
-- Configuring x64-windows-rel
-- Using msys root at C:/Users/Arjan/source/repos/performous/vcpkg/downloads/tools/msys2/aaafe87dcead272e
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:155 (message):
    Command failed: C:/Users/Arjan/source/repos/performous/vcpkg/downloads/tools/msys2/aaafe87dcead272e/usr/bin/make.exe -j 33 --trace -f Makefile all
    Working Directory: C:/Users/Arjan/source/repos/performous/vcpkg/buildtrees/fontconfig/x64-windows-dbg
    See logs for more information:
      C:\Users\Arjan\source\repos\performous\vcpkg\buildtrees\fontconfig\build-x64-windows-dbg-out.log
      C:\Users\Arjan\source\repos\performous\vcpkg\buildtrees\fontconfig\build-x64-windows-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_make.cmake:191 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_make.cmake:26 (vcpkg_build_make)
  ports/fontconfig/portfile.cmake:47 (vcpkg_install_make)
  scripts/ports.cmake:141 (include)


Error: Building package fontconfig:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: fontconfig:x64-windows
  Vcpkg version: 2021-07-26-9425cf5f512f242c0bcbabac31f08832825aee81
```
build-x64-windows-dbg-err.log
```
cl : Command line warning D9002 : ignoring unknown option '-Xcompiler'
cl : Command line warning D9002 : ignoring unknown option '-Xcompiler'
Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30038.1 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

fcobjshash.gperf.h
Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30038.1 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

fcobjshash.gperf.h
cl : Command line warning D9002 : ignoring unknown option '-Xcompiler'
/bin/sh: line 25: /usr/bin/xgettext: No such file or directory
/bin/sh: line 27: /usr/bin/xgettext: No such file or directory
make[3]: *** [Makefile:247: fontconfig.pot-update] Error 127
make[2]: *** [Makefile:314: fontconfig.pot] Error 2
make[1]: *** [Makefile:601: all-recursive] Error 1
make: *** [Makefile:485: all] Error 2
```
[build-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/6947450/build-x64-windows-dbg-out.log)

**Additional context**
This still worked on [`a863c84812089836a3c0f2f215ab3e2579fc8acf`](https://github.com/microsoft/vcpkg/commit/a863c84812089836a3c0f2f215ab3e2579fc8acf) every commit after this it fails with the same logs provided above.

Rerunning the `.\bootstrap-vcpkg.bat` script and rebuilding doesn't solve this sadly

**Comment by JackBoosY** [comment]
Can you please update vcpkg to latest and try again?

**Comment by JonLiu1993** [comment]
@Baklap4 ,Thanks for posting this issue,please updaye your vcpkg to latest

**Comment by Baklap4** [comment]
It seems the latest master resolves this problem. When was this problem resolved? I did try a couple of times since i've noticed this (before filing this issue)

**Comment by JackBoosY** [comment]
@Baklap4 Fixed by #19361.
